File: PyStructs/Arrays/ArrayImplementation.py
class MyArray:
    def __init__(self):
        self.length = 0
        self.data = {}

    def __str__(self):
        return str(self.list())

    def list(self):
        for items in self.data:
            return " ".join(str(self.data[i]) for i in range(self.length))

    def get(self, index):
        return self.data[index]

    def push(self, item):
        self.length += 1
        self.data[self.length-1] = item

    def pop(self):
        last_item = self.data[self.length-1]
        del self.data[self.length-1]
        self.length -= 1
        return last_item

    def remove(self, index):
        item = self.data[index]
        for i in range(index, self.length-1):
            self.data[i] = self.data[i+1]
        del self.data[self.length-1]
        self.length -= 1
        return item

    def insert(self, index, item):
        self.length += 1
        for i in range(self.length-1, index, -1):
            self.data[i] = self.data[i-1]
        self.data[index] = item

File: JSStructs/ArrayImplementation.js
class MyArray {
    constructor() {
        this.length = 0;
        this.data = {};
    }

    get(index) {
        return this.data[index];
    }

    push(item) {
        this.data[this.length] = item;
        this.length++;
    }

    pop() {
        const lastItem = this.data[this.length-1];
        delete this.data[this.length-1];
        this.length--;
        return lastItem;
    }

    delete(index) {
        const item = this.data[index];
        this.shiftItems(index);
        return this.pop();
    }

    shiftItems(index) {
        for (let i = 0; i < this.length; i++) {
            this.data[index] = this.data[index + 1];
        }
    }
}

const newArray = new MyArray();
newArray.push('hi');newArray.push('sup');newArray.push('how are you?');
console.log(newArray);
let x = newArray.delete(0);
console.log(newArray);
console.log(x);

// newArray.pop();
// let x = newArray.pop();
// console.log(newArray);
// console.log(x);
